sdk/src/driftClient.ts

```diff
@@ -6554,24 +6626,36 @@ export class DriftClient {
 			allIx.push(instructions);
 		}
 
+		console.debug(`🔧:: Building TX with computeUnits::${computeUnits} computeUnitsPrice::${computeUnitsPrice}`);
+
 		txVersion = txVersion ?? this.txVersion;
+
+		const latestBlockHashAndContext = await this.connection.getLatestBlockhashAndContext({
+			commitment: this.opts.preflightCommitment,
+		});
+
 		if (txVersion === 'legacy') {
-			return new Transaction().add(...allIx);
+
+			const message = new TransactionMessage({
```

Comment:
THIS is a major kicker that I want you to double check @crispheaney  ... 

When we were using legacy transactions we were also using the "legacy" Transaction class. But you can use the VersionedTransaction class to hold a legacy transaction. The core reason I made this change is because the transaction simulation wouldn't work with the legacy class and this fixed it - but I believe this is an overall improvement. The only thing I'm concerned about is if there are outside consumers of this method which could break now ??? If they were making the same assumption we were (that a VersionedTransaction class means it's a definitely a v0 transaction and not possibly still a legacy one)

Reply:
If this change is a bridge too far then I think I have to uglify the param processor a little bit by like converting the `Transaction` to a `VersionedTransaction` for simulation - but I don't think that's out of the question if you're against this .. this is more of a proposal because it seems like a better long term change

Reply:
tbh im not sure, i think we needed to do this for wallets that didnt support legacy transaction. I think this might cause issues w wallets that dont support versioned tx